### PR TITLE
Docker deployment

### PR DESCRIPTION
**Pull request by louis030195**
usage

```bash
docker run --rm --name flying-squid mineflayer/flying-squid
```

### CODE REVIEW

**Comment by louis030195** [comment]
Work in progress.
Questions:

Do you have a Prismarine Dockerhub ?

**Comment by rom1504** [comment]
just created one

any way to automatically deploy to it using github secrets on new push to master ?

**Comment by rom1504** [comment]
https://docs.docker.com/docker-hub/builds/#service-users-for-team-autobuilds apparently they advise to use another user for that

**Comment by louis030195** [comment]
Yeah I do that here https://github.com/louis030195/niwrad/blob/master/.github/workflows/main.yml

**Comment by rom1504** [comment]
I set up the secrets, it should work.
can you follow https://docs.github.com/en/free-pro-team@latest/actions/guides/publishing-docker-images#publishing-images-to-docker-hub and update the pr with it?
(the org name on dockerhub is prismarinejs)

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          registry: docker.io
+          dockerfile: deploy/Dockerfile
+          tag_with_ref: true
+
+      - name: Push to GitHub Packages
+        uses: docker/build-push-action@v1
+        with:
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+          repository: prismarinejs/flying-squid/flying-squid
```
Comment (louis030195) [inline .github/workflows/docker.yml:28]: Not  sure about this one though, don't know Github Package Docker registry, they say to name "owner/repo/image" unlike DockerHub "owner/image"

**Comment by rom1504** [comment]
@louis030195 can you run `npm run fix` and commit the result to fix the lint ? (that's why CI is not passing)

File: deploy/.dockerignore [hunk 2]
```diff
@@ -0,0 +1,11 @@
+**/node_modules/
```
Comment (rom1504) [inline deploy/.dockerignore:1]: are you sure this should be ignore ?

**Review by rom1504 — APPROVED** [review]
let's try it

**Comment by rom1504** [comment]
@louis030195 didn't work https://github.com/PrismarineJS/flying-squid/runs/1302295058?check_suite_focus=true

can you
1. open a new PR
2. try to fix
3. change the trigger to be on push instead of release so you can iterate directly in the PR like in https://github.com/PrismarineJS/flying-squid/blob/master/.github/workflows/ci.yml#L3

**Comment by louis030195** [comment]
@rom1504 I'm not sure I can push on any registry on my fork which doesn't has  any secrets https://github.com/louis030195/flying-squid/runs/1302399157?check_suite_focus=true#step:5:1

**Comment by rom1504** [comment]
if you open a PR it will run using the main repo secrets I think

**Pull request merged**